osu.Game/Overlays/Toolbar/AnalogClockDisplay.cs

```diff
@@ -70,8 +70,18 @@ private void updateRotation(Drawable hand, float fraction)
 
             float rotation = fraction * 360 - 90;
 
+            // The case where a hand is completing a rotation.
+            // Animate and then move back one full rotation so we don't need to track outside of 0..360
             if (Math.Abs(hand.Rotation - rotation) > 180)
-                hand.RotateTo(rotation);
+            {
+                float animRotation = rotation;
+                while (animRotation < hand.Rotation)
+                    animRotation += 180;
```

Comment:
I found all of this very weird initially (why do all these places use 180 degrees rather than 360? 360deg is what a full rotation is?) but upon closer inspections I guess I understand why (examining the second hand, `hand.Rotation` goes from like 264 degrees to -90 so it's not a full 360 revolution but 59/60ths of one), so I think I'm just going to ignore this because using the "proper" 360 is going to be a pain in the ass anyway to do correctly.

Reply:
yeah. could definitely be rewritten to better logic by not using transforms but maybe a rainy day